test/WebJobs.Script.Tests.Integration/Host/StandbyManagerTests.cs

```diff
@@ -111,16 +112,24 @@ public void IsWarmUpRequest_ReturnsExpectedValue()
                 responseBody = await response.Content.ReadAsStringAsync();
                 Assert.Equal("WarmUp complete.", responseBody);
 
+                // Now specialize the host
+                ScriptSettingsManager.Instance.SetSetting(EnvironmentSettingNames.AzureWebsitePlaceholderMode, "0");
+                request = new HttpRequestMessage(HttpMethod.Get, "api/dne");
+                response = await httpClient.SendAsync(request);
+                Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+
                 httpServer.Dispose();
                 httpClient.Dispose();
 
                 await Task.Delay(2000);
 
                 // verify logs
                 string[] logLines = traceWriter.Traces.Select(p => p.Message).ToArray();
+                string text = string.Join(Environment.NewLine, logLines);
                 Assert.Equal(2, logLines.Count(p => p.Contains("Host is in standby mode")));
                 Assert.Equal(1, logLines.Count(p => p.Contains("Stopping Host")));
                 Assert.Equal(2, logLines.Count(p => p.Contains("Executed 'Functions.WarmUp' (Succeeded")));
+                Assert.Equal(1, logLines.Count(p => p.Contains("Starting host specialization")));
```

Comment:
The logs we emit during specialization should be enough to discern the time spent at various stages. E.g. the output from this test looks like:

```
Function started (Id=f6e5edf3-9f29-426b-98be-2e1efff44769)
Executing 'Functions.WarmUp' (Reason='This function was programmatically called via the host APIs.', Id=f6e5edf3-9f29-426b-98be-2e1efff44769)
WarmUp function invoked.
Function completed (Success, Id=f6e5edf3-9f29-426b-98be-2e1efff44769, Duration=638ms)
Executed 'Functions.WarmUp' (Succeeded, Id=f6e5edf3-9f29-426b-98be-2e1efff44769)
Host lock lease acquired by instance ID '87654639876900123453445678890144'.
Starting host specialization
Reading host configuration file 'C:\Users\mathewc\AppData\Local\Temp\StandbyModeTest\host.json'
Host configuration file read:
{}
Loaded custom extension 'BotFrameworkConfiguration'
Loaded custom extension 'SendGridConfiguration'
Loaded custom extension 'EventGridExtensionConfig'
Generating 0 job function(s)
Starting Host (HostId=mathewcx1-1163069184, Version=1.0.0.0, ProcessId=5364, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. config.UseServiceBus(), config.UseTimers(), etc.).
Job host started
Host lock lease acquired by instance ID '87654639876900123453445678890144'.
Job host stopped
```